### DISCUSSION TITLE
Using WPF conditionally for only Windows builds for a cross platform console application

### DISCUSSION THREAD

**Discussion by ValtteriL** [dotnet/wpf#8034] [Q&A]
I develop a cross-platform console application using .NET7. I want to add GUI to the Windows version of it using WPF.

Is it possible to conditionally include WPF to only Windows builds, while not affecting others? Ideally, I could add a command line flag to the Windows console app, which would start the GUI.

If this is possible, how should it be done?

**Comment by lindexi** [comment]
@ValtteriL Adding the code in csproj when enter your Windows build 

```xml
  <ItemGroup>
    <FrameworkReference Include="Microsoft.WindowsDesktop.App" />
    <FrameworkReference Include="Microsoft.WindowsDesktop.App.WPF" />
  </ItemGroup>
```
